packages/block-library/src/site-logo/edit.js

```diff
@@ -0,0 +1,373 @@
+	// main column, though not infinitely.
+	// @todo It would be good to revisit this once a content-width variable
+	// becomes available.
+	const maxWidthBuffer = maxWidth * 2.5;
+
+	let showRightHandle = false;
+	let showLeftHandle = false;
+
+	/* eslint-disable no-lonely-if */
+	// See https://github.com/WordPress/gutenberg/issues/7584.
+	if ( align === 'center' ) {
+		// When the image is centered, show both handles.
+		showRightHandle = true;
+		showLeftHandle = true;
+	} else if ( isRTL ) {
+		// In RTL mode the image is on the right by default.
+		// Show the right handle and hide the left handle only when it is
+		// aligned left. Otherwise always show the left handle.
+		if ( align === 'left' ) {
+			showRightHandle = true;
+		} else {
+			showLeftHandle = true;
+		}
+	} else {
+		// Show the left handle and hide the right handle only when the
+		// image is aligned right. Otherwise always show the right handle.
+		if ( align === 'right' ) {
+			showLeftHandle = true;
+		} else {
+			showRightHandle = true;
+		}
+	}
+	/* eslint-enable no-lonely-if */
+
+	return (
+		<>
+			<InspectorControls>
+				<PanelBody title={ __( 'Site Logo Settings' ) }>
+					<RangeControl
+						label={ __( 'Image width' ) }
+						onChange={ ( newWidth ) =>
+							setAttributes( { width: newWidth } )
+						}
+						min={ minWidth }
+						max={ maxWidthBuffer }
+						initialPosition={ Math.min(
+							naturalWidth,
+							maxWidthBuffer
+						) }
+						value={ width || '' }
+						disabled={ ! isResizable }
+					/>
+				</PanelBody>
+			</InspectorControls>
+			<ResizableBox
+				size={ { width, height } }
+				showHandle={ isSelected }
+				minWidth={ minWidth }
+				maxWidth={ maxWidthBuffer }
+				minHeight={ minHeight }
+				maxHeight={ maxWidthBuffer / ratio }
+				lockAspectRatio
+				enable={ {
+					top: false,
+					right: showRightHandle,
+					bottom: true,
+					left: showLeftHandle,
+				} }
+				onResizeStart={ onResizeStart }
+				onResizeStop={ ( event, direction, elt, delta ) => {
+					onResizeStop();
+					setAttributes( {
+						width: parseInt( currentWidth + delta.width, 10 ),
+						height: parseInt( currentHeight + delta.height, 10 ),
+					} );
+				} }
+			>
+				{ img }
+			</ResizableBox>
+		</>
+	);
+};
+
+export default function LogoEdit( {
+	attributes,
+	className,
+	setAttributes,
+	isSelected,
+} ) {
+	const { width } = attributes;
+	const [ logoUrl, setLogoUrl ] = useState();
+	const [ error, setError ] = useState();
+	const ref = useRef();
+	const { mediaItemData, sitelogo, url } = useSelect( ( select ) => {
+		const siteSettings = select( 'core' ).getEditedEntityRecord(
+			'root',
+			'site'
+		);
+		const mediaItem = select( 'core' ).getEntityRecord(
+			'root',
+			'media',
+			siteSettings.sitelogo
+		);
+		return {
+			mediaItemData: mediaItem && {
+				url: mediaItem.source_url,
+				alt: mediaItem.alt_text,
+			},
+			sitelogo: siteSettings.sitelogo,
+			url: siteSettings.url,
+		};
+	} );
```

Comment:
Thanks for the update, I believe we should add a `[]` as a second argument here to ensure the callback is not executed on each render.

Reply:
done in 2f08a3b530bfda2b52cf42a44be158a5652ac557